sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -896,7 +897,13 @@ def on_port_update_event(self, port_change_event):
 
             self.force_cmis_reinit(lport, 0)
         else:
+            # PORT_DEL event for the same lport happens 3 times because
+            # we are subscribing to CONFIG_DB, STATE_DB|TRANSCEIVER_INFO, and STATE_DB|PORT_TABLE.
+            # We only handle the first one and ignore the rest.
```

Comment:
Instead of introducing a new attribute `deleted_ports` and its corresponding update logic, you may be able to simply rely on port_change_event's existing `db_name`(/`table_name`), and only proceed for the case of db_name is CONFG_DB 
https://github.com/sonic-net/sonic-platform-daemons/blob/bf865c6b711833347d3c57e9d84cd366bcd1b776/sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py#L30-L31

Reply:
https://github.com/sonic-net/sonic-platform-daemons/pull/500/files#diff-fc70c43977a1ce57f3a6e9d7d09cb7ea5204f52b3adf3162936205e5680efb2fR899-R902

@longhuan-cisco  Thank you for the suggestion. I was able to fix the issue without introducing `deleted_ports`.

I initially added `deleted_ports` because the CmisManagerTask handles port events in two stages.

The first stage is done in `on_port_update_event` via   [`port_change_observer.handle_port_update_event()`](https://github.com/sonic-net/sonic-platform-daemons/pull/500/files#diff-fc70c43977a1ce57f3a6e9d7d09cb7ea5204f52b3adf3162936205e5680efb2fR1313) and the second stage is done after returning from `port_change_observer.handle_port_update_event()` by [`iterating over port_dict`](https://github.com/sonic-net/sonic-platform-daemons/pull/500/files#diff-fc70c43977a1ce57f3a6e9d7d09cb7ea5204f52b3adf3162936205e5680efb2fR1315).

If we delete the port info from `port_dict` in the first stage, the second stage cannot handle the port.
By introducing `deleted_ports`, the origin code was deleting the port info from `port_dict` after finishing the second stage.
However, it looks like currently, the second stage is not doing anything for the removed port. so deleting the port info in the first stage is safe.

Though this works, I think the current two-stage design is bug-prone, because we might want to add some handling(e.g. de-initialize) for the removed port in the second stage.
If you agree, I'd like to refactor the current design in a different PR.